statchatui/src/App/App.jsx

```diff
@@ -23,81 +19,31 @@ function App() {
     platform: "",
     gameName: "",
   });
-  async function fetchStats(formData) {
-    const apexApiKey = process.env.REACT_APP_API_KEY;
-    const baseUrl = "https://api.mozambiquehe.re/bridge?auth=";
-    if (
-      formData.gamertag === "" ||
-      formData.platform === "" ||
-      formData.gameName === ""
-    ) {
-      return null;
-    }
-    try {
-      let response;
-      if (formData.gameName === GAME_NAMES.APEX) {
-        response = await fetch(
-          `${baseUrl}${apexApiKey}&player=${formData.gamertag}&platform=${formData.platform}`
-        );
-      } else if (formData.gameName === GAME_NAMES.R6) {
-        response = await fetch(
-          `https://api.henrikdev.xyz/r6/v1/profile/${formData.platform.toLowerCase()}/${
-            formData.gamertag
-          }`
-        );
-      }
-      const data = await response.json();
-      return data;
-    } catch (error) {
-      console.error("Error submitting form:", error);
-    }
-  }
+
+  const onStatsChange = (statData) => {
+    const newAppData = { ...data };
+    // update newData to have the stats api data
+    newAppData.statData = statData;
+    setData(newAppData);
+  };
+
   useEffect(() => {
     async function fetchData() {
-      try {
-        if (isSignedIn) {
-          const profileResponse = await fetch(
-            `http://localhost:3002/users/profile`,
-            {
-              method: "GET",
-              credentials: "include",
-            }
-          );
-          const profileData = await profileResponse.json();
-          setSignedInUserData(profileData);
-          const stats = await fetchStats(profileData);
-          onStatsChange(stats);
-        }
-      } catch (error) {
-        console.error("Error fetching user info:", error);
+      if (isSignedIn) {
+        const initialData = await fetchInitialData();
+        setSignedInUserData(initialData.profileData);
+        const stats = initialData.stats;
+        onStatsChange(stats);
       }
     }
     fetchData();
   }, [isSignedIn]);
 
   useEffect(() => {
     //everytime the page is refreshed this function will fetch to see if req.session.userId exists, if it does, we will set  signed in to true.
-    const checkSignedIn = async () => {
-      try {
-        const response = await fetch("http://localhost:3002/users/check", {
-          method: "GET",
-          credentials: "include",
-        });
-        const data = await response.json();
-        setIsSignedIn(data.isSignedIn);
```

Comment:
This should be pulled out the same way you did with fetchStats and fetchInitialData.